subprojects/core/src/main/java/org/gradle/util/NameMatcher.java

```diff
@@ -25,8 +25,8 @@
  * Selects a single item from a list of candidates based on a camel-case pattern.
```

Comment:
Don't forget to update the javadoc ;)

Reply:
It's just a wip at this point, but thanks for the pointer.

Reply:
Actually, the javadoc is still valid. The pattern will remain camel case but it will also match to kamel-case strings.

Reply:
I see. It would be nice if it included some notes about what `selects` means in that context.

Reply:
Yes it can be confusing. I think it would be good to add a bit more explanation here maybe with a sample like:

>  Both `camelCase` (matched by e.g. `cC`) and `kebab-case` (matched by e.g. `kC`) candidates can be matched.

Reply:
This sentence might be brief but still correct. So the pattern can be only camel case (like `mAP`). It does match however to camel case (`myAwesomeProject`) and kebab case (`my-awesome-project`) strings.